service/create/load_balancer_test.go

```diff
@@ -0,0 +1,93 @@
+package create
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/giantswarm/awstpr"
+	"github.com/giantswarm/clustertpr"
+	"github.com/giantswarm/clustertpr/cluster"
+	"github.com/juju/errgo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLoadBalancerName(t *testing.T) {
+	tests := []struct {
+		name       string
+		domainName string
+		tpo        awstpr.CustomObject
+		res        string
+		err        error
+	}{
+		{
+			name:       "working",
+			domainName: "component.foo.bar.example.com",
+			tpo: awstpr.CustomObject{
+				Spec: awstpr.Spec{
+					Cluster: clustertpr.Cluster{
+						Cluster: cluster.Cluster{
+							ID: "foo-customer",
+						},
+					},
+				},
+			},
+			res: "foo-customer-component",
+		},
+		{
+			name:       "missing ID key in cloudconfig",
+			domainName: "component.foo.bar.example.com",
+			tpo: awstpr.CustomObject{
+				Spec: awstpr.Spec{
+					Cluster: clustertpr.Cluster{
+						Cluster: cluster.Cluster{
+							ID: "",
+						},
+					},
+				},
+			},
+			res: "",
+			err: missingCloudConfigKeyError,
```

Comment:
There is a check for maleformed formats in `hostedZoneName` which then returns `malformedCloudConfigKeyError`. I would go with the same style here.